spec/factories/physical_storage.rb

```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+  factory :physical_storage,
+          :class => "ManageIQ::Providers::Autosde::StorageManager::PhysicalStorage" do
+    name { "test_physical_storage" }
+  end
+
+  factory :physical_storage_autosde,
+          :parent => :physical_storage do
+    sequence(:ems_ref) { |n| "some-uuid-#{seq_padded_for_sorting(n)}" }
+  end
```

Comment:
Provider specific factories should live in the provider plugin unless they have to be used by core
```suggestion
```